### PR TITLE
Add jest-dom and upgrade jest + testing library

### PR DESCRIPTION
**Pull request by mayank99**
Sequence of steps taken:
1. Added `jest-dom` for new matchers (like `toBeVisible` and `toBeDisabled`).
2. Renamed `setupTests.js` to `setupTests.ts` so that ESM import for jest-dom is transformed by `ts-jest`.
3. Moved setup into `setupFilesAfterEnv` (otherwise jest-dom errors out).
4. Changed jest environment from `node` to `jsdom` (otherwise setupFilesAfterEnv errors out).
5. Finally got tests to run. Fixed failing tests in a few places.
6. Added check for `CSS.supports` in Badge to fix test in Tile.
7. Upgraded jest and testing library to fix remaining failures.

### CODE REVIEW

File: src/core/Select/Select.test.tsx [hunk 1]
```diff
@@ -202,14 +202,14 @@ it('should respect visible prop', () => {
   expect(select).toBeTruthy();
```
Comment (bentleyvk) [inline src/core/Select/Select.test.tsx:203]: Can you try changing our focus checks with this one `toHaveFocus()`?

Comment (mayank99) [reply src/core/Select/Select.test.tsx:203]: Changed in this file

File: src/core/Tile/Tile.test.tsx [hunk 2]
```diff
@@ -37,6 +37,7 @@ it('should render new and selected states', () => {
 });
 
 it('should render main text content correctly', () => {
+  window.CSS = { supports: () => true, escape: (i) => i };
```
Comment (bentleyvk) [inline src/core/Tile/Tile.test.tsx:40]: Why this line is needed? For badges?

Comment (mayank99) [reply src/core/Tile/Tile.test.tsx:40]: yes. I'm surprised it was not failing before

Comment (bentleyvk) [reply src/core/Tile/Tile.test.tsx:40]: Yeah... Maybe it is worth moving to the global setup?

Comment (mayank99) [reply src/core/Tile/Tile.test.tsx]: Removed this line and added `supports?.` to Badge

**Review by bentleyvk — APPROVED** [review]

**Pull request merged**